### ISSUE TITLE
Unicorn 2 M68K: cannot write to SR register using Go bindings

### ISSUE THREAD

**Issue by zznop** [unicorn-engine/unicorn#1501]
I am using Go bindings from `2.0.0-rc4` (and built/installed unicorn from the same tag). The following program demonstrates the issue:

```
package main

import (
	"log"

	uc "github.com/unicorn-engine/unicorn/bindings/go/unicorn"
)

func main() {
	emu, err := uc.NewUnicorn(uc.ARCH_M68K, uc.MODE_BIG_ENDIAN)
	if err != nil {
		log.Fatal(err)
	}

	// Write to D0 to show that it works
	if err := emu.RegWrite(uc.M68K_REG_D0, 0x1337); err != nil {
		log.Fatal(err)
	}

	d0, err := emu.RegRead(uc.M68K_REG_D0)
	if err != nil {
		log.Fatal(err)
	}

	log.Printf("D0: %08x", d0)

	// Write to S0 to show that it doesn't :(
	if err := emu.RegWrite(uc.M68K_REG_SR, 0x1337); err != nil {
		log.Fatal(err)
	}

	sr, err := emu.RegRead(uc.M68K_REG_SR)
	if err != nil {
		log.Fatal(err)
	}

	log.Printf("SR: %08x", sr)
}
```

```
> ./example
2021/11/26 14:21:30 D0: 00001337
2021/11/26 14:21:30 SR: 00000000
```

Moreover, it seems as though unicorn 2 still can't emulate M68K `mov *, sr` instructions. When trying to emulate the following instruction I get a `Unhandled CPU exception (UC_ERR_EXCEPTION)` error.

```
46FC 2700                                         move    #$2700,sr
```

Between unicorn not supporting moves to `SR` and not being able to set my own code hooks to emulate the instruction myself, it makes unicorn unusable for M68K. Perhaps I'm doing something wrong?

**Comment by zznop** [comment]
I've made a separate issue for unhandled CPU exceptions on unsupported M68K instructions after running into more: https://github.com/unicorn-engine/unicorn/issues/1502

**Comment by zznop** [comment]
Submitted a PR to fix this. See https://github.com/unicorn-engine/unicorn/pull/1507

**Comment by wtdcode** [comment]
Closed due to #1502 fixed.

Thanks.
